Tests fail when upgrading glib from 2.22 to 2.24 if sys-devel/gdb is installed
because gdb is run on .libs/assert-msg-test before LD_LIBRARY_PATH is set. This
causes gdb to use the system-wide glib instead, and fail on the test.

This patch exports LD_LIBRARY_PATH before running gdb

https://bugzilla.gnome.org/621368

(updated for 2.31.18)

--- tests/run-assert-msg-test.sh
+++ tests/run-assert-msg-test.sh
@@ -34,7 +34,7 @@
        msg_test="lt-$msg_test"
 fi
 echo_v "Running gdb on assert-msg-test"
-OUT=$(gdb --batch --ex run --ex "set print elements 0" --ex "print (char*) __glib_assert_msg" .libs/$msg_test 2> $error_out) || \
+OUT=$(libtool --mode=execute gdb --batch --ex run --ex "set print elements 0" --ex "print (char*) __glib_assert_msg" .libs/$msg_test 2> $error_out) || \
   fail "failed to run gdb"
 
 echo_v "Checking if assert message is in __glib_assert_msg"
